src/config.ts

```diff
@@ -98,6 +98,10 @@ const StorybookShotsSchema = z.object({
    * [ 320, 768, 1280 ]
    */
   breakpoints: z.array(z.number()).optional(),
+  /**
+   * Target specific element on page with a selector
+   */
+  elementLocator: z.string().optional(),
```

Comment:
Nice feature @pedrosousa13.
I have a question/suggestion:
What do you think about turning this into an array of element locators (like masks).
I can imagine situations where you have multiple elements on a page that you'd like to screenshot.
For that, each selector would need a label that would be used to give each shot name a unique suffix.
what do you think?

```suggestion
  elementLocator: z.object({
    label: z.string(),
    selector: z.string(),
  }).optional(),
```

Reply:
Hi Chris, I thought about doing that but the locator element doesn't accept arrays: https://playwright.dev/docs/api/class-page#page-locator

we would have to take multiple screenshots, 1 per array element. Maybe it's overkill?

This is mostly a case for design systems where 1 page = 1 component. People can still override this per story in storybook though via the parameters.

You can still use the selector to select multiple elements with 1 string via css selectors though. Not sure how playwright handles it if they're in different places which is interesting.

What do you think? @chriskalmar

Reply:
I think you are right when saying that DS usually has a single component in place. As we aim to more or less centralise the features across modes probably it will be more useful for pages mode where you would make screenshots of particular elements on the page. It will be of course additional screenshots. So essentially if you have a landing page & want to just screenshot a pricing section & your CTA than you would provide those two selectors directly to the page as params.

In DS you could only have some carthesian stories that cover many different variants of component & you would want to screenshot only couple of variants for some reason. That's where the array might be useful for storybook mode. 

Still open for the discussion here.

Reply:
I think it can make sense! We make a story for each variant that's why I didn't think of it.

Not sure how taking multiple screenshots in this shots file would affect other systems and names of shots. I think it would take more exploration. 

How about this, let's merge this as a string for now and add the array possibility in another iteration? I can start working on that immediately after this one 😄

Reply:
Sounds good 👍 
Taking multiple shots on a single Story would indeed require a bit of refactoring here: 
https://github.com/lost-pixel/lost-pixel/blob/main/src/crawler/storybook.ts#L313

It would be similar to how we deal with breakpoints, but yes, let's move this to a separate issue.

Reply:
awesome. good that you already have the breakpoints. I'll take a look at that logic to make the element locator array 😄

Reply:
if nothing else is necessary, can we merge this one? 😄